code/modules/admin/sql_notes.dm

```diff
@@ -0,0 +1,226 @@
+	var/output
+	var/navbar
+	var/ruler
+	ruler = "<hr style='background:#000000; border:0; height:3px'>"
+	navbar = "<a href='?_src_=holder;nonalpha=1'>\[All\]</a>|<a href='?_src_=holder;nonalpha=2'>\[#\]</a>"
+	for(var/letter in alphabet)
+		navbar += "|<a href='?_src_=holder;shownote=[letter]'>\[[letter]\]</a>"
+	navbar += "<br><form method='GET' name='search' action='?'>\
+	<input type='hidden' name='_src_' value='holder'>\
+	<input type='text' name='notessearch' value='[index]'>\
+	<input type='submit' value='Search'></form>"
+	if(!linkless)
+		output = navbar
+	if(target_ckey)
+		var/target_sql_ckey = sanitizeSQL(target_ckey)
+		var/DBQuery/query_get_notes = dbcon.NewQuery("SELECT id, timestamp, notetext, adminckey, last_editor, server FROM [format_table_name("notes")] WHERE ckey = '[target_sql_ckey]' ORDER BY timestamp")
+		if(!query_get_notes.Execute())
+			var/err = query_get_notes.ErrorMsg()
+			log_game("SQL ERROR obtaining ckey, notetext, adminckey, last_editor, server from notes table. Error : \[[err]\]\n")
+			return
+		output += "<h2><center>Notes of [target_ckey]</center></h2>"
+		if(!linkless)
+			output += "<center><a href='?_src_=holder;addnote=[target_ckey]'>\[Add Note\]</a></center>"
+		output += ruler
+		while(query_get_notes.NextRow())
+			var/id = query_get_notes.item[1]
+			var/timestamp = query_get_notes.item[2]
+			var/notetext = query_get_notes.item[3]
+			var/adminckey = query_get_notes.item[4]
+			var/last_editor = query_get_notes.item[5]
+			var/server = query_get_notes.item[6]
+			output += "<b>[timestamp] | [server] | [adminckey]</b>"
+			if(!linkless)
+				output += " <a href='?_src_=holder;removenote=[id]'>\[Remove Note\]</a> <a href='?_src_=holder;editnote=[id]'>\[Edit Note\]</a>"
+				if(last_editor)
+					output += " <font size='2'>Last edit by [last_editor] <a href='?_src_=holder;noteedits=[id]'>(Click here to see edit log)</a></font>"
+			output += "<br>[notetext]<hr style='background:#000000; border:0; height:1px'>"
+	else if(index)
+		var/index_ckey
+		var/search
+		output += "<center><a href='?_src_=holder;addnoteempty=1'>\[Add Note\]</a></center>"
+		output += ruler
+		if(!isnum(index))
+			index = sanitizeSQL(index)
+		switch(index)
+			if(1)
+				search = "^."
+			if(2)
+				search = "^\[^\[:alpha:\]\]"
+			else
+				search = "^[index]"
+		var/DBQuery/query_list_notes = dbcon.NewQuery("SELECT DISTINCT ckey FROM [format_table_name("notes")] WHERE ckey REGEXP '[search]' ORDER BY ckey")
+		if(!query_list_notes.Execute())
+			var/err = query_list_notes.ErrorMsg()
+			log_game("SQL ERROR obtaining ckey from notes table. Error : \[[err]\]\n")
+			return
+		while(query_list_notes.NextRow())
+			index_ckey = query_list_notes.item[1]
+			output += "<a href='?_src_=holder;shownoteckey=[index_ckey]'>[index_ckey]</a><br>"
+	else
+		output += "<center><a href='?_src_=holder;addnoteempty=1'>\[Add Note\]</a></center>"
+		output += ruler
+	usr << browse(output, "window=show_notes;size=900x500")
```

Comment:
You might try using the /datum/browser stuff for this, it tends to make things look nice even with weird HTML.